src/controllers/chat.controller.js

```diff
@@ -0,0 +1,122 @@
+require('dotenv').config();
+const path = require('path');
+
+const db = require(path.join(__dirname, '../db/index'));
+const queries = require(path.join(__dirname, '../db/queries'));
+const getPayload = require(path.join(__dirname, '../helpers/jwt'));
+
+const addUser = async (userId, chatId) => {
+    try {
+        await db.none(queries.addUserToChat, [userId, chatId]);
+    }
+    catch(error) {
+        return Promise.reject('User already member of chat');
+    }
+}
+
+exports.createChat = async (req, res, next) => {
+    try {
+        // name is optional
+        const name = req.body.name || null;
+        
+        // create chat; return id
+        const { id: chatId } = await db.one(queries.createChat, [name]);
+        
+        const { id: userId } = getPayload(req.headers);
+        // add user who created the chat to the chat
+        await addUser(userId, chatId);
+        
+        // get chat just added to table, searching by id returned 
+        const chat = await db.one(queries.findChatById, [chatId]);
+
+        // created chat; return chat id and name
+        return res.status(201).json(chat);
+    }
+    catch(error) {
+        if (error) return next(error);
+    }
+}
+
+exports.createMessage = async (req, res, next) => {
+    try {
+        if (!req.body.text) {
+            return next('Missing fields');
+        }
+
+        const { id: userId } = getPayload(req.headers);
+        // message values: chat id, user id and text
+        const message = [
+            req.params.cid,
+            userId,
+            req.body.text
```

Comment:
I wonder if we want these fields to exactly match up with the Redux state, guess it would make sense to make them mirror each other as much as possible.

Reply:
Manipulating the data so it's in the format expected on the frontend is atypical from what I've heard. Usually the data is retrieved in whatever format from the service on the frontend and the proper transformations are made to the data to suit the frontend/views. If there's anything major that you feel should be changed though, let me know, @mnicole. 

Also, I just corrected some typos/missing info that I made earlier in the apidoc comments - thanks for catching a couple, so I was able to take a quick look again and correct all - I think - of them, @ernewtoner.

Reply:
Also not sure what you mean by these fields specifically? This actually references the db design, which we worked on for the project plan and during the first week of work (week 3), where the message table has columns for chat ID (`chats_id`), user ID (`users_id`) and text (`text`). `created_at` defaults to the present time. Changing the naming conventions at this point would be a bit arduous, but it's possible, if we find that that's seriously necessary.  What would have been better is looking at the db design and using that to format the frontend design...

Reply:
Interesting, yeah that makes sense, seems like data is always kind of decoupled like that and doesn't need to match up sequentially in any way.

Ah no worries about the comments--figured you'd catch them anyway once you looked at it again.

Reply:
Just saw your 2nd comment (page wasn't refreshed)--was just wondering in general, definitely no need to change the db design. Yeah I'm more thinking about the Redux design, could revisit it in those respects.

Reply:
I thought you might've checked that when you were setting up redux, but I didn't check your PR to verify. It's unfortunate if you find you have to make a lot of modifications. I'm afraid it wouldn't make a lot of sense to format the data in a specific way on the backend contrary to the DB design just for the frontend. It would've made/makes more sense to design the frontend accordingly. That would've been something crucial to look at for the redux design, I'm afraid.

Reply:
If that's something that's a major roadblock, we should discuss/I can help with whatever. This is why communication and looking at all the information available, like the DB design that was already decided upon early on, is super important.

Reply:
I did look at the db when doing redux state and it matches it mostly--with maybe the one mistake of using username instead of user id--sorry if my comment was misleading, I was just making a general comment about matching them up cause I was wondering if fields being sequentially matched woulb e a good thing (which we addressed in the above comments), didn't mean to directly comment on or criticize the backend implementation. The Redux state can be modified easily right now, so it's no big deal if we want to make modifications.

Reply:
No worries, I didn't think you were criticizing the backend; I was just responding to your inquiry. It's not just the db design but other aspects of the backend that would have been important to consider. We can discuss more about how to proceed at the meeting today, so we can avoid this sort of lapse in communication and design in the future.